sway-launcher-desktop.sh

```diff
@@ -19,6 +20,8 @@ if [[ "$1" == 'describe' ]]; then
   exit
 fi
 
+TERMINAL_COMMAND="termite -e"
```

Comment:
I wonder if we can do something with `$TERM` and a small switch/case that returns the correct argument for common terminal applications. I will try to find out how other solutions are dealing with this. Being able to pass this as an option in the future is probably still a good idea. Until that is possible, I am not so sure if it's agood idea to hardcode this, however. Maybe this needs to be done first.

Reply:
Problem is `$TERM` is not meant to be executable. In a lot of cases it’s just `xterm-256color` (I came across [this discussion of why `$TERM` won’t work](https://bugs.freedesktop.org/show_bug.cgi?id=92514) when researching this)

There’s also an `xdg-terminal` tool that is in the same project as `xdg-open` and seemed promising until I saw that:

1. It’s not included in the `xdg-utils` package on arch
2. It has a hard coded fallback to xterm if you aren’t using a desktop environment (this would include us using Sway standalone)

So the best idea I could come up with was making it easy to change in this script for now, and then making it configurable down the road.

I’m all about a better solution though if someone knows of one!

Reply:
Thank you for putting in the time to research this. It's a shame `$TERM` is useless for this purpose and no real alternative exists. An environment variable does look like the smartest thing to do then, yeah

Reply:
Can't we just execute the command directly? This script already runs in the user-selected terminal emulator, after all. So why not simply re-use that instead? I think we could still make use of an env var as an override (since you may want to use different parameters/window class/whatever), but

Reply:
The reason I did this is because running desktop files with `Terminal=true` doesn't work currently. Selecting an option like "ranger" just closes the launcher and does nothing. This is likely because `swaymsg exec ranger` does nothing. I did a little more digging and found [that `wofi` recently added support for this as well](https://hg.sr.ht/~scoopta/wofi/rev/33cb9e5529c1fe2239224996f446e60331da3d13) (a little different, it's not looking at the `Terminal` option in the desktop entry, but instead using shift+enter as the indication that the user wants to open in a terminal). In order to use the feature you have to specify a terminal command in the config or pass it in with the `--terminal` option. So it seems the precedence is there to do it this way.

Reply:
Made a couple updates here. Added a comment describing how to change this value and made it so the env var is only assigned if it doesn't already have a value.

Reply:
Well my idea was to not use `swaymsg exec` in that case, but rather call the command directly within the script. I guess this would work, but has the downside that the launcher window might be too small to be useful for any serious TUI program, so you'd essentially be forced to use an override anyway.